compiler/rustc_pattern_analysis/src/pat.rs

```diff
@@ -147,6 +147,21 @@ impl<Cx: PatCx> fmt::Debug for DeconstructedPat<Cx> {
     }
 }
 
+/// Delegate to `uid`.
+impl<Cx: PatCx> PartialEq for DeconstructedPat<Cx> {
+    fn eq(&self, other: &Self) -> bool {
+        self.uid == other.uid
+    }
+}
+/// Delegate to `uid`.
+impl<Cx: PatCx> Eq for DeconstructedPat<Cx> {}
+/// Delegate to `uid`.
+impl<Cx: PatCx> std::hash::Hash for DeconstructedPat<Cx> {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.uid.hash(state);
+    }
+}
```

Comment:
Because of `covered_by: FxHashSet<&'p DeconstructedPat<Cx>>`. There's no map from uids to patterns so I can't store just the uids. I could have used the derived `Hash`, it would just have been much worse perf.

Reply:
Yeah ok, just an optimization is what I was curious about.